### DISCUSSION TITLE
select an offered completion by number?

### DISCUSSION THREAD

**Discussion by bkerin** [akinomyoga/ble.sh#284] [Q&A] [answered]
Manual section 7.2.7 contains this tantalizing part:  

bleopt menu_linewise_prefix=' \e[1;32m%2d:\e[m '

I've never used indirect selection but it seems like it might be better than my current habit of typing more until disambiguous (since that's often sort of unknown how much be typed and the file is often visually recognized immediately).  But I didn't see anyway to go from these label numbers to selection by number?

**Comment by akinomyoga** [comment]
> But I didn't see anyway to go from these label numbers to selection by number?

Right. Currently, there is no way to select the menu item by numbers. The feature is originally implemented for the general uses of the menu interface, and I haven't initially intended to utilize it for the completion interface. But maybe selecting by numbers is also useful for the completion menu.

**Comment by akinomyoga** [comment, marked as answer]
> I've never used indirect selection but it seems like it might be better than my current habit of typing more until disambiguous (since that's often sort of unknown how much be typed and the file is often visually recognized immediately). But I didn't see anyway to go from these label numbers to selection by number?

@bkerin I have supported this in commit b91b8bc8a8d44148095568a1cd2bde897724e8dd. You can use <kbd>M-0</kbd>...<kbd>M-9</kbd> to start selecting a menu item by its index (immediately after starting the completion or when in the `menu-complete` mode). If you would like to also use <kbd>0</kbd>..<kbd>9</kbd> to start the selection by the index, you can use the following settings:

```bash
# blerc

for key in {0..9}; do
  ble-bind -m emacs -f "$key" 'append-arg enter-menu'
  ble-bind -m vi_imap -f "$key" 'append-arg enter-menu'
  ble-bind -m menu_complete -f "$key" 'menu/append-arg always'
done
```

I have also added options [`bleopt menu_prefix`](https://github.com/akinomyoga/ble.sh/wiki/Manual-%C2%A77-Completion#user-content-bleopt-menu_prefix), `bleopt menu_align_prefix`, `bleopt menu_desc_prefix`, and `bleopt menu_dense_prefix`. These can be used to set the prefix of the menu items for the respective menu styles. `menu_prefix` is the common default.

**Reply by akinomyoga** [reply]
@bkerin Also, I'd appreciate it if you could try and test this.

**Comment by bkerin** [comment]
Seems to work great :)

 I haven't tested default binding as M-number is all window manager for me and it being dwm I have to rebuild to keep it from intercept so a hassle.

I didn't search the entire manual for where the binding approach (for key in {0..9} etc.) is mentioned but I wouldn't have known how to bind this without the explicit instructions.

**Reply by akinomyoga** [reply]
Thank you! It's sufficient to test it with the `for key in {0..9}` bindings. I'd add the example later somewhere in the manual.
